pytorch/launch_nv.py

```diff
@@ -85,157 +85,200 @@
-  '--batch-sched', '192,192,128',
-  '--resize-sched', '0,18,41',
-  '--epochs', 45,
-  '--lr', 0.4,
-  '--dist-url', 'file:///home/ubuntu/data/file.sync', # single instances are faster with file sync
+  '--lr-phases', [
+    {'ep':0,  'sz':128, 'bs':192, 'trndir':'-sz/160'},
+    {'ep':(0,5),  'lr':(0,lr)},
+    {'ep':5,      'lr':lr},
+    {'ep':18, 'sz':224, 'bs':192},
+    {'ep':20,     'lr':lr/10},
+    {'ep':34,     'lr':lr/100},
+    {'ep':41, 'sz':288, 'bs':128, 'min_scale':0.5},
+    {'ep':(42,45),'lr':lr/1000}
+  ],
+  '--no-autoscale-lr2batch',
   '--num-tasks', 1,
-  '--ami-name', 'Deep Learning AMI (Ubuntu) Version 12.0'
+  '--ami-name', 'Deep Learning AMI (Ubuntu) Version 12.0',
+  '--dist-url', 'file:///home/ubuntu/data/file.sync', # single instances are faster with file sync
 ]
 
 
 # Current best settings
 # Current benchmark for 1x p3
+lr = 1.0
 xar_args = [
-  '--lr-sched', '4,15,26,32',
-  '--batch-sched', '512,192,128',
-  '--resize-sched', '0,14,32',
-  '--epochs', 35,
-  '--lr', 1.0,
-  '--dist-url', 'file:///home/ubuntu/data/file.sync', # single instances are faster with file sync
+  '--lr-phases', [
+    {'ep':0,  'sz':128, 'bs':512, 'trndir':'-sz/160'},
+    {'ep':(0,5),  'lr':(lr,lr*2)}, # custom lr warmup works better with init-bn0 flag
+    {'ep':5,      'lr':lr},
+    {'ep':14, 'sz':224, 'bs':192},
+    {'ep':16,     'lr':lr/10},
+    {'ep':27,     'lr':lr/100},
+    {'ep':32, 'sz':288, 'bs':128, 'min_scale':0.5, 'use_ar':True},
+    {'ep':(33,35),'lr':lr/1000}
+  ],
   '--init-bn0',
+  '--no-bn-wd',
   '--num-tasks', 1,
-  '--val-ar',
-  '--lr-linear-scale',
   '--ami-name', 'Deep Learning AMI (Ubuntu) Version 12.0',
-  '--no-bn-wd',
-  '--env-name', 'pytorch_source_tv'
+  '--env-name', 'pytorch_source'
+  '--dist-url', 'file:///home/ubuntu/data/file.sync', # single instances are faster with file sync
 ]
 
+lr = 1.0
 xar_args_pytorch = [
-  '--lr-sched', '4,15,26,32',
-  '--resize-sched', '0,14,32',
-  '--batch-sched', '512,192,128',
-  '--epochs', 35,
-  '--lr', 1.0,
-  '--dist-url', 'file:///home/ubuntu/data/file.sync', # single instances are faster with file sync
+  '--phases', [
+    {'ep':0,  'sz':128, 'bs':512, 'trndir':'-sz/160'},
+    {'ep':(0,5),  'lr':(lr,lr*2)},
+    {'ep':5,      'lr':lr},
+    {'ep':14, 'sz':224, 'bs':192},
+    {'ep':16,     'lr':lr/10},
+    {'ep':27,     'lr':lr/100},
+    {'ep':32, 'sz':288, 'bs':128, 'min_scale':0.5, 'use_ar':True},
+    {'ep':(33,35),'lr':lr/1000}
+  ],
   '--init-bn0',
+  '--no-bn-wd',
   '--num-tasks', 1,
-  '--val-ar',
-  '--lr-linear-scale',
   '--ami-name', 'pytorch.imagenet.source.v5',
-  # '--ami-name', 'Deep Learning AMI (Ubuntu) Version 12.0',
+  # '--env-name', 'pytorch_p36',
   '--env-name', 'pytorch_source',
-  '--no-bn-wd',
-  '--c10d'
+  '--dist-url', 'file:///home/ubuntu/data/file.sync', # single instances are faster with file sync
+  # '--c10d'
 ]
 
-# 2x p3's - is this still being used?
+
+# 2x p3 for throughput benchmarking
+lr = 1.0
```

Comment:
I've 1.5x'd the batch size here, so I scaled the learning rate accordingly. Sorry, I didn't know that we were still using this, otherwise I wouldn't have changed anything. Feel free to revert these changes

Reply:
ah ok, larger lr makes sense then